src/test/scala/is/hail/expr/ir/MatrixIRSuite.scala

```diff
@@ -255,4 +255,15 @@ class MatrixIRSuite extends SparkSuite {
     assert(rows < 20 && rows > 0)
     assert(entries < 400 && entries > 0)
   }
+
+  @Test def testMatrixAggregateColsByKeyWithEntriesPosition() {
+    val range = MatrixTable.range(hc, 3, 3, Some(1)).ast
+    val withEntries = MatrixMapEntries(range, MakeStruct(FastIndexedSeq("x" -> 2)))
+    val m = MatrixAggregateColsByKey(
```

Comment:
Can you confirm this replicated the error and your fix resolved it?

Reply:
yeah. I wrote the test first to make sure it failed.